Add modals for properties with valueTemplateRefs

ndushay opened the pull request:
To review:
- take a careful look at existing components I *changed*:
    - ResourceTemplate  (replaced FormWrapper with ResourceTemplateForm, improved tests)
    - sinopiaServerSpoof  (added barcode rt to quiet console errors during testing)
    - MonographInstance (changed barcode rtId prefix)
- look at the new classes carefully
     - ResourceTemplateForm renders
     - ModalToggle renders
     - ResourceTemplateModal 
- think about whether I am testing the new classes sufficiently

If there's objection to the removal of unused components, I can take those commits out of this PR.

Known problems/todos:
- integration tests (#223)
- need to accommodate these this:  `repeatable` (#217), `mandatory` (#218), `editable` and `defaultValue` (#219 for both)
- use propTypes.shape for more complex objects in arrays (part of #211)
- ~~ensure valueTemplateRefs are always resourceTemplate ids (and not vocabularies, etc) (#220)~~
- holding the data from a ResourceTemplateModal  (once we decide how we will be modelling that:  redux or ...) (#221)
- styling of buttons, etc:  (#222)

Editor spins up like this:

![image](https://user-images.githubusercontent.com/96775/49264853-cc557800-f404-11e8-8d0b-cfe719729303.png)

and if you click a button (e.g. "Title"), you get something like this:

![image](https://user-images.githubusercontent.com/96775/49264872-d9726700-f404-11e8-878b-13c56ca8d340.png)

and if you click the "Title note" button, you get this:

![image](https://user-images.githubusercontent.com/96775/49264886-e5f6bf80-f404-11e8-924b-512b7d7df046.png)


Accomplished by replacing FormWrapper component with ResourceTemplateForm component, which is reused in the Modals.

SaravShah commented on src/components/editor/ResourceTemplateModal.jsx:
```diff
@@ -0,0 +1,57 @@
+  }
+
+  saveDataFromModal = () => {
+    console.log('TODO: implement holding data from ResourceTemplateModal')
```
Should we create a ticket # for this?

jermnelson replied:
Yeah, I think this is a candidate for using Redux.

SaravShah commented on src/components/editor/ModalToggle.jsx:
```diff
@@ -0,0 +1,51 @@
+  toggleModal = () => {
+    const prevState = this.state
+    prevState.visible = !prevState.visible
+    this.setState(prevState => (prevState))
```
i think you could also do this if you want: 

```
this.setState({
    visible: !this.state.visible
})
```

ndushay replied:
I tried some stuff;   dunno if I tried this.

SaravShah commented on __tests__/components/editor/ResourceTemplateForm.test.js:
```diff
@@ -0,0 +1,94 @@
+      "type": "lookup"
+    },
+    {
+      "propertyLabel": "Bullseye!",
```
loving the `propertyLabels`

jermnelson commented on static/spoofedFilesFromServer/fromSinopiaServer/resourceTemplates/MonographInstance.json:
```diff
@@ -175,7 +175,7 @@
         },
```
This is not specific to this code, but I realized another source we should use for determining such variables as `repeatable` and `mandatory` is the Library of Congress Bibframe documentation at https://www.loc.gov/bibframe/. For example, in this BIBFRAME file the 'instanceOf` propertyTemplate has false for repeatable, but I as understand BIBFRAME 2.0 ontology, at this level of modeling, Instance should have only one Work, while Works have multiple Instances.

jgreben commented on __tests__/components/editor/ResourceTemplateForm.test.js:
```diff
@@ -0,0 +1,94 @@
+    })
+    it('passes appropriate props to ModalToggle(s)', () => {
+      wrapper.find('ModalToggle').forEach((node) => {
+        expect(node.prop('buttonLabel')).toEqual('Note') // TODO: ensure we get from spoofed resourceTemplate
```
Maybe say a bit more about this TODO or issue#?

ndushay replied:
haha - I have no idea what I was thinking when I wrote that TODO comment.  I will remove it.

ndushay commented:
Josh, et al:

yes, sorry about the length.  At least I removed as many files as I added - heh.

I made a separate ticket for the integration tests(#223);  hopefully this is one of the largest PRs we'll have to do for a long time.  (cough cough).   I think whenever possible we should include integration tests with the PR ... if we think the UI is settled down enough.  I already made a ticket for the "backlog" of missing integration tests (#208): and Sarav made a ticket for the InputLiteral (#212), so ... ahem.

 :-)
